Assets/Scripts/Internal/DaggerfallMobileBillboard.cs

```diff
@@ -0,0 +1,734 @@
+        }
+
+        void Start()
+        {
+            if (Application.isPlaying)
+            {
+                // Get component references
+                mainCamera = GameObject.FindGameObjectWithTag("MainCamera").GetComponent<Camera>();
+                meshFilter = GetComponent<MeshFilter>();
+                meshRenderer = GetComponent<MeshRenderer>();
+            }
+        }
+
+        void OnDisable()
+        {
+            restartAnims = true;
+        }
+
+        void Update()
+        {
+            // Restart animation coroutine if not running
+            if (restartAnims)
+            {
+                StartCoroutine(AnimateEnemy());
+                restartAnims = false;
+            }
+
+            // Rotate to face camera in game
+            if (mainCamera && Application.isPlaying)
+            {
+                // Rotate billboard based on camera facing
+                cameraPosition = mainCamera.transform.position;
+                Vector3 viewDirection = -new Vector3(mainCamera.transform.forward.x, 0, mainCamera.transform.forward.z);
+                transform.LookAt(transform.position + viewDirection);
+
+                // Orient enemy based on camera position
+                UpdateOrientation();
+            }
+        }
+
+        /// <summary>
+        /// Sets new enemy type.
+        /// </summary>
+        /// <param name="dfUnity">DaggerfallUnity singleton. Required for content readers and settings.</param>
+        /// <param name="enemyType">Enemy type.</param>
+        public override void SetEnemy(DaggerfallUnity dfUnity, MobileEnemy enemy, MobileReactions reaction, byte classicSpawnDistanceType)
+        {
+            // Initial enemy settings
+            summary.Enemy = enemy;
+            summary.EnemyState = MobileStates.Move;
+            summary.Enemy.Reactions = reaction;
+
+            // Load enemy content
+            int archive = GetTextureArchive();
+            CacheRecordSizesAndFrames(dfUnity, archive);
+            AssignMeshAndMaterial(dfUnity, archive);
+
+            // Apply enemy state and update orientation
+            lastOrientation = -1;
+            ApplyEnemyState();
+
+            // Set initial contact range
+            summary.ClassicSpawnDistanceType = classicSpawnDistanceType;
+
+            // Raise setup flag
+            summary.IsSetup = true;
+        }
+
+        /// <summary>
+        /// Sets a new enemy state and restarts frame counter.
+        /// Certain states are one-shot only (such as attack and hurt) and return to idle when completed.
+        /// Continuous states (such as move) will keep looping until changed.
+        /// </summary>
+        /// <param name="state">New state.</param>
+        /// <returns>Frames per second of new state.</returns>
+        public override float ChangeEnemyState(MobileStates state)
```

Comment:
Return value is never used and specific to billboard animation. Can i remove it?

Reply:
Apologies for not answering. If never used, happy for this return value to be removed.